tool/menu.lua

```diff
@@ -148,36 +148,55 @@ end
 
 local function randomizermenu()
   local selected = ui.select({
+    "New Seed on New Game    " .. randomizer.newgamenewseedui[randomizer.newgamenewseed],
     "Proficiency: " .. randomizer.proficiency,
-    "Unseeded",
-    "Seeded",
+    "Random seed",
+    "Set seed",
     "Reset",
     "Back",
   })
-  if selected == 1 then
+  if selected == 1 then -- New Seed on New Game
+    local index = table.indexof(randomizer.newgamenewseedvalues, randomizer.newgamenewseed)
+    index = ((index - 1 + 1) % #randomizer.newgamenewseedvalues) + 1
+    randomizer.newgamenewseed = randomizer.newgamenewseedvalues[index]
+
+  elseif selected == 2 then -- Proficiency
     local index = table.indexof(randomizer.proficiencies, randomizer.proficiency)
     index = ((index - 1 + 1) % #randomizer.proficiencies) + 1
     randomizer.proficiency = randomizer.proficiencies[index]
-  elseif selected == 2 then
+
+  elseif selected == 3 then -- Unseeded
+    randomizer.seedqueue = {}
     randomizer.seed = ""
```

Comment:
Given the `seedqueue` and using `resetfunction` to call `randomizer.randomize`, I think it would make sense to set `randomizer.seed` within the `randomize()` function and demote it to a local. That reduces duplication here and mental overhead.

Reply:
The randomizer.seed was needed here in the first place because the UI needs a seed to display before resetfunction is called for the first time. I'll think of a way around that though.

Reply:
I added a randomizer.hudisfornextseed boolean that solves both this issue and helps with the levelindex one below.